src/main/java/carpet/mixins/NoiseColumnSamplerMixin_scarpetMixin.java

```diff
@@ -0,0 +1,235 @@
+
+    @Shadow @Final
+    private InterpolatedNoiseSampler terrainNoise;
+
+    @Shadow @Final @Nullable
+    private SimplexNoiseSampler islandNoise;
+
+    @Shadow @Final
+    private DoublePerlinNoiseSampler aquiferBarrierNoise;
+
+    @Shadow @Final
+    private DoublePerlinNoiseSampler aquiferFluidLevelFloodednessNoise;
+
+    @Shadow @Final
+    private DoublePerlinNoiseSampler aquiferFluidLevelSpreadNoise;
+
+    @Shadow @Final
+    private DoublePerlinNoiseSampler aquiferLavaNoise;
+
+    @Shadow @Final
+    private DoublePerlinNoiseSampler pillarRarenessNoise;
+
+    @Shadow @Final
+    private DoublePerlinNoiseSampler pillarThicknessNoise;
+
+    @Shadow @Final
+    private DoublePerlinNoiseSampler spaghetti2dElevationNoise;
+
+    @Shadow @Final
+    private DoublePerlinNoiseSampler spaghetti2dModulatorNoise;
+
+    @Shadow @Final
+    private DoublePerlinNoiseSampler spaghetti2dThicknessNoise;
+
+    @Shadow @Final
+    private DoublePerlinNoiseSampler spaghetti3dFirstNoise;
+
+    @Shadow @Final
+    private DoublePerlinNoiseSampler spaghetti3dSecondNoise;
+
+    @Shadow @Final
+    private DoublePerlinNoiseSampler spaghetti3dRarityNoise;
+
+    @Shadow @Final
+    private DoublePerlinNoiseSampler spaghetti3dThicknessNoise;
+
+    @Shadow @Final
+    private DoublePerlinNoiseSampler spaghettiRoughnessModulatorNoise;
+
+    @Shadow @Final
+    private DoublePerlinNoiseSampler caveCheeseNoise;
+
+    @Shadow @Final
+    private DoublePerlinNoiseSampler oreGapNoise;
+
+    @Shadow @Final
+    private GenerationShapeConfig config;
+
+    @Shadow
+    protected abstract double method_38409(double d, double e, double f);
+
+    @Shadow
+    protected abstract double sampleCaveLayerNoise(int x, int y, int z);
+
+    @Shadow
+    protected abstract double samplePillarNoise(int x, int y, int z);
+
+    @Shadow
+    protected abstract double sampleSpaghetti2dNoise(int x, int y, int z);
+
+    @Shadow
+    protected abstract double sampleSpaghetti3dNoise(int x, int y, int z);
+
+    @Shadow
+    protected abstract double sampleSpaghettiRoughnessNoise(int x, int y, int z);
+
+    @Shadow
+    protected abstract double sampleCaveEntranceNoise(int x, int y, int z);
+
+    @Shadow
+    public abstract double sampleShiftNoise(int x, int y, int z);
+
+    @Shadow
+    private static double sample(DoublePerlinNoiseSampler sampler, double x, double y, double z, double invertedScale) {
+        throw new AssertionError();
+    }
+
+    @Shadow
+    public abstract MultiNoiseUtil.NoiseValuePoint sample(int i, int j, int k);
+
+    @Shadow
+    public abstract TerrainNoisePoint createTerrainNoisePoint(int x, int z, float continentalness, float weirdness, float erosion, Blender blender);
+
+    @Override
+    public double getNoiseSample(String name, int x, int y, int z) {
+        MultiNoiseUtil.NoiseValuePoint noiseValuePoint = this.sample(x, y, z);
+        switch (name) {
+            case "temperature" -> {
```

Comment:
not sure what's better, a giant switch, or a string keyed map with lambda calls.

Reply:
I just find switch much more readable. Is there a performance difference between using map over this giant switch ? If so I can switch it over.

Reply:
ig using a map would allow you to move it about where you're gonna use it more, whereas with a switch it's only here in a mixin, where people may not instinctively go to find it. Plus the mixin folder is huge, so finding the right one could be a pain.

Reply:
true. I will give it a go